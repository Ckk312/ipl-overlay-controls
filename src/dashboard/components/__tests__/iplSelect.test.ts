import IplSelect from '../iplSelect.vue';
import { mount } from '@vue/test-utils';

describe('IplSelect', () => {
    it('applies given label to element', () => {
        const wrapper = mount(IplSelect, { props: { label: 'Label!', modelValue: '', options: []} });

        expect(wrapper.get('label').text()).toEqual('Label!');
    });

    it('applies given options to select element', () => {
        const wrapper = mount(IplSelect, { props: { modelValue: '', options: [
            { name: 'Option', value: 'opt' },
            { name: 'Option the Second', value: 'opt2' },
            { name: 'Opt 3', value: 'optthree' }
        ]} });

        expect(wrapper.get('select').html()).toMatchSnapshot();
    });

    it('emits update message on change', async () => {
        const wrapper = mount(IplSelect, { props: { modelValue: '', options: [
            { name: 'Option', value: 'opt' },
            { name: 'Option the Second', value: 'opt2' },
            { name: 'Opt 3', value: 'optthree' }
        ]} });

        await wrapper.get('select').setValue('opt2');

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted.length).toEqual(1);
        expect(emitted[0]).toEqual(['opt2']);
    });
});
