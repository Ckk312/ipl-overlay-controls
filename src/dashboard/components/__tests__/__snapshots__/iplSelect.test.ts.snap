// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplSelect applies given options to select element 1`] = `"<select><option value=\\"opt\\">Option</option><option value=\\"opt2\\">Option the Second</option><option value=\\"optthree\\">Opt 3</option></select>"`;
